### PR TITLE
Account sortcode numbers

### PR DESCRIPTION
**Pull request by adrianlio**
- Updated Account class constructors to support sort codes and account numbers
- NewBank.java now includes hardcoded sort code that is assigned to every account
- NewBank.java has new 'assignAccountNumber' function which checks if a randomly generated account number is already in use and if not, returns a new account number

### CODE REVIEW

**Review by AlexMxwl — CHANGES_REQUESTED** [review]
This works and the way it has been implemented makes sense, however there is a _very_ small chance of generating the same account number more than once.

It would be a good idea to record all of the account numbers that get assigned and then perform a check to see whether a new account number is the same as any of those (and if so, just run the random number generator again until you get a unique account number).

File: newbank/server/NewBank.java [hunk 1]
```diff
@@ -150,12 +175,35 @@ private boolean isNumeric(String string) {
 		return Pattern.matches(regex, string);
 	}
 
+	private int assignAccountNumber() {
+		Random r = new Random();
+		//generate a random account number that is not in current list
+		while (accountNumberList.contains(accountNumberCurrent)) {
```
Comment (adrianlio) [inline newbank/server/NewBank.java:181]: Hi @AlexMxwl! If you could double check, this is the bit where in the current code it should be checking the current list of assigned accounts. When NewBank is created, a private static ArrayList of integers 'accountNumberList' is created that stores account numbers (as Integers) that have been assigned out. This 'while' loop should keep running and generate new numbers if the generated account number is found in that array list. I could try to run a loop on the actual account numbers stored in customer accounts but thought that would be quite inefficient each time as opposed to using an ArrayList.contains function 

I suppose I should have probably coded something incase we run out of numbers though which could actually be a likely scenario in a real bank... :D Once we reach 10,000,000 customers the server will hang...

**Review by adrianlio — COMMENTED** [review]
Hi @AlexMxwl ! If you could see my comment on NewBank.java row 181, hopefully that answers your question? I believe this should be doing this check already but please do correct me if it's not working that way. Thanks!

File: newbank/server/NewBank.java (hunk 1 above)
Comment (AlexMxwl) [reply newbank/server/NewBank.java:181]: Ah, you're absolutely right, sorry that I missed this! I've followed the logic through now and I think the way you've implemented it is sensible.

**Pull request merged**